sdk/core/src/policies/retry_policies/linear_retry.rs

```diff
@@ -0,0 +1,76 @@
+use crate::policies::{Context, Policy, PolicyResult, Request, Response};
+use crate::sleep::sleep;
+use chrono::{DateTime, Local};
+use std::sync::Arc;
+use std::time::Duration;
+
+/// Retry policy with linear backoff.
+///
+/// Retry policy with linear backoff (with an added random delay up to 256 ms). Each retry will
+/// happen at least after the same, configured sleep time. The policy will retry until the maximum number of
+/// retries have been reached or the maximum allowed delay has passed (whichever comes first). The
+/// wait time is not precise.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct LinearRetryPolicy {
+    delay: Duration,
+    max_retries: u32,
+    max_delay: Duration,
+}
+
+impl Default for LinearRetryPolicy {
+    fn default() -> Self {
+        Self {
+            delay: Duration::from_secs(3),
+            max_retries: 3,
+            max_delay: Duration::from_secs(10),
+        }
+    }
+}
+
+impl LinearRetryPolicy {
+    fn is_expired(
+        &self,
+        first_retry_time: &mut Option<DateTime<Local>>,
+        current_retries: &u32,
+    ) -> bool {
+        if *current_retries > self.max_retries {
+            return true;
+        }
+
+        if first_retry_time.is_none() {
+            *first_retry_time = Some(Local::now());
+        }
+
+        Local::now()
+            > first_retry_time.unwrap() + chrono::Duration::from_std(self.max_delay).unwrap()
+    }
+}
+
+#[async_trait::async_trait]
+impl Policy for LinearRetryPolicy {
+    async fn send(
+        &self,
+        ctx: &mut Context,
+        request: &mut Request,
+        next: &[Arc<dyn Policy>],
+    ) -> PolicyResult<Response> {
+        let mut first_retry_time = None;
+        let mut current_retries = 0;
+
+        loop {
+            match next[0].send(ctx, request, &next[1..]).await {
+                Ok(response) => return Ok(response),
```

Comment:
So I see a policy does get a response, but how would a policy create a response to short-circuit the pipeline and not call `next`? For e.g., to proxy calls.

Reply:
The policy can just create a response (not sure how easy that is currently but there's nothing preventing us from making it easy) and return that without calling `next[0].send`.

Reply:
To make this easier I've added a simple struct that implements `Stream` based on a byte buffer (called `BytesStream`). A policy that wants to short-circuit the pipeline can then construct a response easily (I've used it in the mock wasm implementation for reqwest).